src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceSourceConfig.java

```diff
@@ -161,9 +161,8 @@ public boolean isSoqlQuery() {
                                      SalesforceSourceConstants.PROPERTY_QUERY);
   }
 
-  @Override
   public void validate(FailureCollector collector) {
-    super.validate(collector);
+    this.getConnection().validate(collector);
```

Comment:
nit: instead of `this.getConnection()`, just use `connection`

Reply:
This change has been done as asked by you, but some test cases in SalesforceSourceConfigTest are eventually failing after this change. As we need to mock the validation of real connection while writing the test cases but after this change, it is not possible to mock that call.
this.getConnection().validate(collector) , this enables us to mock the validation. But the change asked by you does not.
This change needs to be rollback to make those test cases work. May I ?
Example test case which is failing is : testPKChunkWithChunkSizeBelowMin() in SalesforceSourceConfigTest class

Reply:
ok